apps/aecore/lib/aecore/chain/chainstate.ex

```diff
@@ -77,27 +94,23 @@ defmodule Aecore.Chain.Chainstate do
     AccountStateTree.root_hash(chainstate.accounts)
   end
 
-  def filter_invalid_txs(txs_list, chainstate, block_height) do
-    {valid_txs_list, _} =
-      List.foldl(txs_list, {[], chainstate}, fn tx, {valid_txs_list, chainstate_acc} ->
-        {valid_chainstate, updated_chainstate} = validate_tx(tx, chainstate_acc, block_height)
-
-        if valid_chainstate do
-          {[tx | valid_txs_list], updated_chainstate}
-        else
-          {valid_txs_list, chainstate_acc}
-        end
-      end)
-
-    Enum.reverse(valid_txs_list)
-  end
-
-  @spec validate_tx(SignedTx.t(), Chainstate.t(), non_neg_integer()) ::
-          {boolean(), Chainstate.t()}
-  defp validate_tx(tx, chainstate, block_height) do
-    {true, apply_transaction_on_state!(tx, chainstate, block_height)}
-  catch
-    {:error, _} -> {false, chainstate}
+  @doc """
+  Goes through all the transactions and only picks the valid ones
+  """
+  @spec get_valid_txs(list(), Chainstate.t(), non_neg_integer()) :: list()
+  def get_valid_txs(txs_list, chainstate, block_height) do
+    txs_list
+    |> List.foldl([], fn tx, valid_txs_list ->
```

Comment:
(important) If an transaction is valid next transaction should be executed on updated chainstate.

Explenation:
Transactions can have dependencies between each other. For example let's say there was created a two transactions using the same funds. They both are valid independently, but together they aren't. This function is supposed to return first one, because we use this function to filter transactions so we can create a valid block.